nautobot/core/forms/fields.py

```diff
@@ -714,6 +716,78 @@ def has_changed(self, initial, data):
         return super().has_changed(initial, data)
 
 
+class JSONArrayChoiceFormField(JSONArrayFormField):
+    """
+    A FormField counterpart to JSONArrayField that supports choice validation.
```

Comment:
I'm curious - was there a reason this needed to be a new subclass versus just directly enhancing `JSONArrayFormField` to have an optional `choices` parameter?

Reply:
Yep, do to the different prepare_value (and possibly widget).